### PR TITLE
DOP-4960: Updated refresh on search page dark mode

### PR DESCRIPTION
**Pull request by biancalaube**
### Stories/Links:

DOP-4960
### Current Behavior:

[current search results](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/seung.park/DOP-4928/search/index.html?q=and&page=1)
^ this is not really the current behaviour on prod, this is Seung's staging link for DOP-4928. But in this link you can see that when the page refreshes on dark mode some components do light mode on the refresh.
### Staging Links:

[staging link with updated ui on reload of search page](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-4960/search/index.html?q=and)

### Notes:
- Before these updates, if you were on dark mode and the page refreshed the skeleton, drop shadow, tags and text would flash light mode first before doing dark mode. With these updates, when the search page is refreshed it should not flash light if on dark mode.
- For viewing, since the staging link you can't refresh the page. Just retype the index.html in the URL since it disappears on load:
amazonaws.com/master/landing/bianca.laube/DOP -4960/search/ _**index.html**_ ?q=and
- I also didn't touch the light mode for the filters on the side since matt was working on those for another ticket
- If you are having trouble viewing the difference if you run it locally its slower and easier to see the changes for refresh

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Comment by mmeigs** [comment]
It looks like the spacing between the Title and the search result count is off. Maybe just need to pull in the latest changes from Seung's branch.

Also, could you also work on the Button that is only shown on mobile for search filters?

**Comment by biancalaube** [comment]
> It looks like the spacing between the Title and the search result count is off. Maybe just need to pull in the latest changes from Seung's branch.
> 
> Also, could you also work on the Button that is only shown on mobile for search filters?

@mmeigs these 2 updates should be updated in my newest commit, i also updated the staging link so hopefully that works now :)

File: src/components/SearchResults/SearchResult.js [hunk 1]
```diff
@@ -90,17 +99,62 @@ const searchResultLinkStyling = ({ searchResultTitleColor, searchResultTitleColo
...
+
+  .dark-theme & {
+    background-color: ${palette.blue.dark3};
+    border: 1px solid ${palette.blue.dark2};
+    color: ${palette.blue.light1};
+  }
+`;
+
+const styledTagPurple = css`
+  background-color: ${palette.purple.light3};
```
Comment (mmeigs) [inline src/components/SearchResults/SearchResult.js:147]: nit: For these css (vs styled) declarations, we usually have a naming convention of `___styles`

File: src/components/SearchResults/SearchResults.js [hunk 2]
```diff
@@ -211,6 +228,26 @@ const StyledLoadingSkeletonContainer = styled('div')`
...
+    background: linear-gradient(110deg, ${palette.gray.light2} 35%, ${palette.gray.light3}, ${palette.gray.light2} 65%)
+      0 0/ 100vw 100% fixed;
+    background-position: 50vw 0;
+    animation: SkeletonShimmer 1.5s infinite linear;
+
+    @keyframes SkeletonShimmer {
+      to {
+        background-position: 100vw 0;
+      }
+    }
```
Comment (mmeigs) [inline src/components/SearchResults/SearchResults.js:242]: I'm wondering if you need these lines at all

File: src/components/SearchResults/SearchResult.js (hunk 1 above)
Comment (biancalaube) [reply src/components/SearchResults/SearchResult.js:147]: sorry matt, can you link what the naming convention should be for __styles, Is that a document ? Like do you want me to rename the variable ?

File: src/components/SearchResults/SearchResults.js (hunk 2 above)
Comment (biancalaube) [reply src/components/SearchResults/SearchResults.js:242]: These lines are from the leafy green code, it was makes the skeleton shimmer, without it when i changed the background color then the skeleton stopped shimmering all together without this code. However, I can still edit it if you like!

File: src/components/SearchResults/SearchResult.js (hunk 1 above)
Comment (mmeigs) [reply src/components/SearchResults/SearchResult.js:147]: So, if you use `css`, we usually name it something like `purpleTagStyles`. But if we use `styled`, we usually name it `StyledPurpleTag`. Just renaming!

File: src/components/SearchResults/SearchResults.js (hunk 2 above)
Comment (mmeigs) [reply src/components/SearchResults/SearchResults.js:242]: Hmm, I'd say try again. I just pulled down the branch and took these lines out and the shimmer is still there!

File: src/components/SearchResults/SearchResult.js [hunk 3]
```diff
@@ -3,14 +3,12 @@ import sanitizeHtml from 'sanitize-html';
...
 import { palette } from '@leafygreen-ui/palette';
 import { Body } from '@leafygreen-ui/typography';
-import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
 import { css, cx } from '@leafygreen-ui/emotion';
 import { theme } from '../../theme/docsTheme';
 import Tag, { searchTagStyle, tagHeightStyle } from '../Tag';
 import SearchContext from './SearchContext';
 import { getFacetTagVariant } from './Facets/utils';
 import { searchResultDynamicStyling } from './SearchResults';
-import { SEARCH_THEME_STYLES } from './styles/searchThemeStyles';
```
[the remark this answers is not in the document]
Comment (biancalaube) [reply src/components/SearchResults/SearchResult.js:13]: Deleted !

File: src/components/SearchResults/SearchResult.js [hunk 4]
```diff
@@ -90,17 +99,62 @@ const searchResultLinkStyling = ({ searchResultTitleColor, searchResultTitleColo
...
   ${({ maxLines }) => truncate(maxLines)};
   // Reserve some space inside of the search result card when there is no preview
   min-height: 20px;
+
+  > span {
+    .dark-theme & {
+      background-color: ${palette.green.dark2};
+    }
+
+    background-color: ${palette.green.light2};
```
[the remark this answers is not in the document]
Comment (biancalaube) [reply src/components/SearchResults/SearchResult.js]: thanks for the input, added a comment but let me know if you want a different comment :)

File: src/components/SearchResults/SearchResult.js (hunk 1 above)
Comment (biancalaube) [reply src/components/SearchResults/SearchResult.js]: Sorry yes that makes sense, i made that change now.

File: src/components/SearchResults/SearchResults.js (hunk 2 above)
Comment (biancalaube) [reply src/components/SearchResults/SearchResults.js]: Sorry yes, that works for me now! resolved.

File: src/components/SearchResults/SearchResult.js [hunk 5]
```diff
@@ -106,20 +106,21 @@ const StyledPreviewText = styled(Body)`
   // Reserve some space inside of the search result card when there is no preview
   min-height: 20px;
 
+  // Changes the highlight color of text in the cards of the search term
```
Comment (mmeigs) [inline src/components/SearchResults/SearchResult.js:109]: We can just say "Targets highlighted search term"

**Review by mmeigs — APPROVED** [review]
LGTM! Make sure to change the comment and also pull in all the latest changes from `DOP-4616` and make sure test snapshots get updated if necessary!

File: src/components/SearchResults/SearchResult.js [hunk 6]
```diff
@@ -61,24 +59,35 @@ const StyledResultTitle = styled('p')`
...
+const searchResultLinkStyling = css`
   height: 100%;
   text-decoration: none;
   border-radius: ${theme.size.medium};
 
   ${StyledResultTitle} {
-    color: ${searchResultTitleColor};
+    color: ${palette.blue.base};
+
+    .dark-theme & {
```
Comment (seungpark) [inline src/components/SearchResults/SearchResult.js:70]: can possibly move these to the Tag component, but not a blocker

**Review by seungpark — APPROVED** [review]
this LGTM. PTAL at the merge conflicts and let me know if there are any differences!

File: AWSCLIV2.pkg
Comment (seungpark) [inline AWSCLIV2.pkg:1]: not sure why this file was added cc @mmeigs deleting from branch DOP-4616

**Pull request merged**